public/content/en/gems/bit-manipulation.md

```diff
@@ -0,0 +1,136 @@
+# Bit manipulation
+
+An excellent example of D's ability to generate code on compile-time with mixins,
+is bit manipulation.
+
+### Simple bit manipulation
+
+D offers the following operator for bit manipulation:
+
+- `&` bitwise and
+- `|` bitwise or
+- `~` bitwise negative
+- `<<`  bitwise signed   left-shift
+- `>>`  bitwise signed   right-shift (preserves the sign of the high-order bit)
+- `>>>` bitwise unsigned right-shift
+
+If you are unfamiliar with binary numbers, you might want to take a short
```

Comment:
> I would rather not introduce bit operations in detail here. Anyone who is interested in bit operations should know them already. But the overview of available operations is great

Fair point, I did mostly did this because optional freshups are often quite useful, e.g. most people hear about bit manipulation in uni and then forget most parts.

After thinking a bit about it, I agree with you and will remove this part.